homeassistant/components/watchyourlan/config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Config flow for WatchYourLAN integration."""
+
+import logging
+
+import voluptuous as vol
+from watchyourlanclient import WatchYourLANClient
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SSL
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# Schema for the user setup form
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=8840): int,
+        vol.Optional(CONF_SSL, default=False): bool,
+    }
+)
+
+
+class WatchYourLANConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WatchYourLAN."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                # Validate input and construct the URL
+                proto = "https" if user_input[CONF_SSL] else "http"
+                target = f"{proto}://{user_input[CONF_HOST]}:{user_input[CONF_PORT]}"
+
+                # Use the WatchYourLANClient to validate the connection
+                api_client = WatchYourLANClient(base_url=target, async_mode=True)
+                hosts = await api_client.get_all_hosts()
+                if not hosts:
+                    raise CannotConnect  # noqa: TRY301
```

Comment:
It's complaining about raising an error inside a `try`. However, this seems like a time that really does make it more user-friendly to raise the "CannotConnect" instead of whatever the api_client might throw.

Reply:
You can set errors, and then change the `else` of the `try` block to `if not errors`